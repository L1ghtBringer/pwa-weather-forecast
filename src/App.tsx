import React from 'react'
import Forecast from 'pages/Forecast/Forecast'
import { AppWrapper, LoveMercury } from './App.styles'

const App: React.FC = () => {
  return <AppWrapper>
    <Forecast/>
    <LoveMercury>C ЛЮБОВЬЮ ОТ MERCURY DEVELOPMENT</LoveMercury>
  </AppWrapper>
}

export default App
