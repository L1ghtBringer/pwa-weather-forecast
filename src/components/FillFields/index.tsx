import React from 'react'
import { FillFieldsContainer } from './fillfields.styles'
import cloud from './icons/cloud.svg'

const FillFields: React.FC = () => {
  return (
    <FillFieldsContainer>
      <img src={cloud} alt="cloud-icon" />
      <h2>Fill in all the fields and the weather will be displayed</h2>
    </FillFieldsContainer>
  )
}

export default FillFields
