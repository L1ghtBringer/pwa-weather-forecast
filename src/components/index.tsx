export { default as Calendar } from './Calendar'
export { default as DatePicker } from './DatePicker'
export { default as Select } from './Select'
export { default as FillFields } from './FillFields'
export { default as WeatherCard } from './WeatherCard'
export { default as Carousel } from './Carousel'
export { default as Snackbar } from './Snackbar'
