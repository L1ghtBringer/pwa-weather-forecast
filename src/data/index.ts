export const cities = [
  {
    value: {
      latitude: 55.796127,
      longitude: 49.106405
    },
    label: 'Kazan'
  },
  {
    value: {
      latitude: 45.03547,
      longitude: 38.975313
    },
    label: 'Krasnodar'
  },
  {
    value: {
      latitude: 53.195873,
      longitude: 50.100193
    },
    label: 'Samara'
  },
  {
    value: {
      latitude: 51.533557,
      longitude: 46.034257
    },
    label: 'Saratov'
  },
  {
    value: {
      latitude: 53.507836,
      longitude: 49.420393
    },
    label: 'Tolyatti'
  }
]
