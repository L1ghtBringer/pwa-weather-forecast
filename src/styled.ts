import baseStyled, { ThemedStyledInterface } from 'styled-components'

export const theme = {
  media: {
    laptop: '(max-width: 1366px)',
    tablet: '(max-width: 805px)',
    tabletS: '(max-width: 700px)',
    phone: '(max-width: 520px)'
  },
  color: {
    accent: '#373AF5',
    secondary: {
      active: '#8083A4',
      hover: '#8083A499'
    },
    baseStong: '#2C2D76',
    critic: '#E5596D',
    baseWeak: '#FFFFFF'
  },
  font: 'Ubuntu, sans-serif'
}

export type Theme = typeof theme

export const styled = baseStyled as ThemedStyledInterface<Theme>
