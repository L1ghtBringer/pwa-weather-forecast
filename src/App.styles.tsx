import styled from 'styled-components'

export const AppWrapper = styled.div`
  width: 100%;
  min-height: 100vh;
  height: auto;
  position: relative;
`

export const LoveMercury = styled.p`
  color: #FFFFFF;
  opacity: 0.6;
  font-family: Ubuntu, sans-serif;
  font-size: 14px;
  position: absolute;
  left: calc(50% - 140px);
  bottom: 16px;
`
