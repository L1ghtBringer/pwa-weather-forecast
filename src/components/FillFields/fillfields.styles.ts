import styled from 'styled-components'

export const FillFieldsContainer = styled.div`
  margin-top: 64px;
  text-align: center;
  img {
    width: 160px;
    height: 160px;
  }
  h2 {
    margin: 24px 0 37px;
    font-size: 16px;
    line-height: 24px;
    color: #8083a4;
  }
  @media ${(props) => props.theme.media.tabletS} {
    margin-top: 44px;
    h2 {
      margin: 24px 0 14px;
    }
  }
  @media ${(props) => props.theme.media.phone} {
    margin-top: 39px;
  }
`
