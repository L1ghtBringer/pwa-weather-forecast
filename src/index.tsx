import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import backgroundDown from 'assets/backgroundDown.svg'
import backgroundDownLow from 'assets/backgroundDownLow.svg'
import backgroundUp from 'assets/backgroundUp.svg'
import backgroundUpLow from 'assets/backgroundUpLow.svg'
import { theme } from './styled'
import './font.css'

const Global = createGlobalStyle`
*,
::after,
::before{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
body {
  overflow-x:hidden;
  margin-right: calc(-1 * (100vw - 100%));
  background: url(${backgroundUp}) top center no-repeat, url(${backgroundDown}) bottom center no-repeat #373af5;;
  background-size: 100%, 100%, cover;
  @media ${theme.media.phone}{
    background: url(${backgroundUpLow}) top center no-repeat, url(${backgroundDownLow}) bottom center no-repeat #373af5;
    background-size: 100%, 100%, cover;
  }
}
`

ReactDOM.render(
  <React.StrictMode>
    <ThemeProvider theme={theme}>
      <Global />
      <App />
    </ThemeProvider>
  </React.StrictMode>,
  document.getElementById('root')
)
