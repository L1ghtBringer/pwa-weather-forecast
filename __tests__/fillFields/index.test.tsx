import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FillFields from '../../src/components/FillFields'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../src/styled'

configure({ adapter: new Adapter() })

const shallowWithTheme = (children: any) =>
  shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

describe('FillFields component test', () => {
  it('Should match the snapshot', () => {
    const weatherCard = shallowWithTheme(<FillFields />)
    expect(weatherCard.html()).toMatchSnapshot()
    expect(weatherCard.exists()).toBe(true)
  })
})
