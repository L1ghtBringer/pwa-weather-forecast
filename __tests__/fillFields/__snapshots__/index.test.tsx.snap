// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FillFields component test Should match the snapshot 1`] = `"<div class=\\"sc-bdnxRM byISJT\\"><img src=\\"[object Object]\\" alt=\\"cloud-icon\\"/><h2>Fill in all the fields and the weather will be displayed</h2></div>"`;
